src/sinks/splunk_hec/logs/sink.rs

```diff
@@ -24,9 +29,9 @@ pub struct HecLogsSink<S> {
     pub source: Option<Template>,
     pub index: Option<Template>,
     pub indexed_fields: Vec<OwnedValuePath>,
-    pub host_key: Option<OwnedTargetPath>,
+    pub host_key: Option<OptionalTargetPath>,
```

Comment:
It might be worth a comment on this struct why an `Option` is needed around the `OptionalTargetPath`s, as those are `Option`s internally as well.